src/Init/Data/List/BasicAux.lean

```diff
@@ -12,60 +12,138 @@ namespace List
   | a::_,  0   => some a
   | _::as, n+1 => get? as n
   | _,     _   => none
 
-def getD (as : List α) (idx : Nat) (a₀ : α) : α :=
-  (as.get? idx).getD a₀
+/--
+Returns the `i`-th element in the list (zero-based).
+
+If the index is out of bounds (`i ≥ as.length`), this function returns `a₀`.
+Also see `get?` and `get!`.
+-/
+def getD (as : List α) (i : Nat) (a₀ : α) : α :=
+  (as.get? i).getD a₀
+
+/--
+Returns the first element in the list.
 
+If the the list is empty, this function panics when executed, and returns `default`.
+See `head` and `headD` for safer alternatives.
+-/
 def head! [Inhabited α] : List α → α
   | []   => panic! "empty list"
   | a::_ => a
 
+/--
+Returns the first element in the list.
+
+If the the list is empty, this function returns `.none`.
+Also see `headD` and `head!`.
+-/
 def head? : List α → Option α
   | []   => none
   | a::_ => some a
 
-def headD : List α → α → α
+/--
+Returns the first element in the list.
+
+If the the list is empty, this function returns `a₀`.
+Also see `head?` and `head!`.
+-/
+def headD : (as : List α) → (a₀ : α) → α
   | [],   a₀ => a₀
   | a::_, _  => a
 
+/--
+Returns the first element of a non-empty list.
+-/
 def head : (as : List α) → as ≠ [] → α
   | a::_, _ => a
 
+/--
+Drops the first element of the list.
+
+If the the list is empty, this function panics when executed, and returns the empty list.
+See `tail` and `tailD` for safer alternatives.
+-/
 def tail! : List α → List α
   | []    => panic! "empty list"
   | _::as => as
 
+/--
+Drops the first element of the list.
+
+If the the list is empty, this function returns `.none`.
+Also see `tailD` and `tail!`.
+-/
 def tail? : List α → Option (List α)
   | []    => none
   | _::as => some as
 
+/--
+Drops the first element of the list.
+
+If the the list is empty, this function returns `as₀`.
+Also see `head?` and `head!`.
+-/
 def tailD : List α → List α → List α
   | [],   as₀ => as₀
   | _::as, _  => as
 
+/--
+Returns the last element of a non-empty list.
+-/
 def getLast : ∀ (as : List α), as ≠ [] → α
   | [],       h => absurd rfl h
   | [a],      _ => a
   | _::b::as, _ => getLast (b::as) (fun h => List.noConfusion h)
 
+/--
+Returns the last element in the list.
+
+If the the list is empty, this function panics when executed, and returns `default`.
+See `getLast` and `getLastD` for safer alternatives.
+-/
 def getLast! [Inhabited α] : List α → α
   | []    => panic! "empty list"
   | a::as => getLast (a::as) (fun h => List.noConfusion h)
 
+/--
+Returns the last element in the list.
+
+If the the list is empty, this function returns `.none`.
+Also see `getLastD` and `getLast!`.
+-/
 def getLast? : List α → Option α
   | []    => none
   | a::as => some (getLast (a::as) (fun h => List.noConfusion h))
 
-def getLastD : List α → α → α
+/--
+Returns the last element in the list.
+
+If the the list is empty, this function returns `a₀`.
+Also see `getLast?` and `getLast!`.
+-/
+def getLastD : (as : List α) → (a₀ : α) → α
```

Comment:
The subscript name seems less useful than a descriptive name here. What about:
```suggestion
/--
Returns the last element in the list.

If the the list is empty, this function returns `fallback`.
Also see `getLast?` and `getLast!`.
-/
def getLastD : (as : List α) → (fallback : α) → α
```
If a very short name is desired, `d` for "default" connects it to the function name, at the risk of confusion with `Inhabited.default`.